data/blog/webview2.md

```diff
@@ -0,0 +1,108 @@
+---
+title: WebView2 and Electron
+author:
+- electron
+date: '2021-07-08'
+---
+
+Over the past week, we’ve received several questions about the differences between the new [WebView2](https://docs.microsoft.com/en-us/microsoft-edge/webview2/) and Electron.
+We want to provide some early information on the similarities and differences of the two solutions.
+
+---
+
+## Architecture Overview
+
+Electron and WebView2 both build from the Chromium source for rendering web content.
+Strictly speaking, WebView2 builds from the Edge source, but Edge is built using a fork of the Chromium source.
+Neither Electron nor WebView2 share any DLL files on the system with Edge or Chrome.
+
+Electron apps always bundle and distribute the exact version of Electron with which they were developed.
+WebView2 has two options in distribution.
+You can bundle the exact WebView2 library your application was developed with, or you can use a shared-runtime version that may already be present on the system.
+WebView2 provides tools for each approach, including a bootstrapping installer in case the shared runtime is missing.
+
+Applications that bundle their frameworks are responsible for updating those frameworks, including minor security releases.
+For apps using the shared WebView2 runtime, WebView2 has its own updater, similar to Chrome or Edge, that runs independent of your application.
+Your app will transparently inherit any updates every time it launches.
+Neither Electron nor WebView2 is managed by Windows Update.
+
+Both Electron and WebView2 inherit Chromium’s multi-process architecture - namely, a single main process that communicates with one-or-more renderer processes.
+These processes are entirely separate from other applications running on the system.
+Whether you run multiple Electron apps or multiple WebView2 apps, each application will contain a whole copy of the following process architectures:
```

Comment:
One caveat - if WebView2 apps use the same user data folder (like a suite of apps would do), then everything except render processes will be shared.  Maybe not a detail that has to be captured here, but want to at least call it out

Reply:
It's worth including. In the interest of helping the end-user, do have a preferred link expanding on user data folders? I ~google'd~ _binged_ and zero MS-links appeared in my first page of results. I'd rather link to the source

Reply:
this is our "best practices" doc around user data folders - does that seem helpful here? https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/user-data-folder